view/src/service.rs

```diff
@@ -256,19 +256,22 @@ impl ViewProtocolService for ViewService {
         for action in tx.actions() {
             if let penumbra_transaction::Action::Spend(spend) = action {
                 let nullifier = spend.body.nullifier;
-                let spendable_note_record = self.storage.note_by_nullifier(nullifier, false).await;
-
-                if spendable_note_record.is_err() {
-                    spend_nullifiers.insert(nullifier, None);
-                } else if let Ok(spendable_note_record) = spendable_note_record {
-                    spend_nullifiers.insert(nullifier, Some(spendable_note_record.note));
+                // An error here indicates we don't know the nullifier, so we omit it from the Perspective.
+                if let Ok(spendable_note_record) =
+                    self.storage.note_by_nullifier(nullifier, false).await
+                {
+                    spend_nullifiers.insert(nullifier, spendable_note_record.note);
```

Comment:
This simplifies the `Perspective` data modeling: either the nullifier is known, in which case it's included, or it's not, in which case it isn't, but we don't record a sentinel `None` value we'd later discard.